### PR TITLE
[webui][api] Refactor RequestSourceDiff mixin

### PR DESCRIPTION
**Pull request by ChrisBr**
and extracted
* BsRequestAction::Differ::ForSource
* BsRequestAction::Differ::QueryBuilder
* BsRequestAction::Differ::QueryBuilderForAccepted
* BsRequestAction::Differ::SourcePackageFinder
* BsRequestAction::Differ

classes.
Also added specs for the new extracted classes.

### CODE REVIEW

File: src/api/spec/models/bs_request_action/differ/source_package_finder_spec.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      context 'and without source access' do
+        before do
+          create(:sourceaccess_flag, project: source_project)
+          login(user)
+        end
+
+        it 'does raise an exception' do
+          expect do
+            finder.all
+          end.to raise_error(Package::ReadSourceAccessError)
```
Comment (bgeuken) [inline src/api/spec/models/bs_request_action/differ/source_package_finder_spec.rb]: Nitpick: I would prefer this to be written in one line

**Comment by hennevogel** [comment]
Phew, next time please make smaller commits...

**Comment by hennevogel** [comment]
and looks like you need to regenerate some cassettes

**Comment by ChrisBr** [comment]
> Phew, next time please make smaller commits...

I actually had it in smaller commits and squashed it as
- the commits were not really useful alone
- wanted to have everything together in case of reverting it, it makes it a lot easier

**Pull request merged**